erpnext/hr/doctype/salary_slip/salary_slip.js

```diff
@@ -156,7 +156,8 @@ var total_work_hours = function(frm, dt, dn) {
 			}
 			gross_pay += earning.amount;
 		});
-		frm.set_value('gross_pay', gross_pay);
```

Comment:
This should not be removed. What if the do_not_include_in_total is unchecked? You should check the do_not_include_in_total condition at line no 153